cobra/core/solution.py

```diff
@@ -296,7 +296,11 @@ def get_solution(model, reactions=None, metabolites=None):
         forward = rxn.forward_variable.name
         reverse = rxn.reverse_variable.name
         fluxes[i] = var_primals[forward] - var_primals[reverse]
-        reduced[i] = var_duals[forward] - var_duals[reverse]
+        try:
+            reduced[i] = var_duals[forward] - var_duals[reverse]
+        except TypeError:
+            # reduced costs are not always defined, e.g. for integer problems
+            reduced[i] = nan
```

Comment:
So better ask for forgiveness than permission? :laughing: I'm fine with it than.